http-clients/src/test/java/com/palantir/remoting/http/SlashEncodingContractTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.http;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import com.squareup.okhttp.mockwebserver.RecordedRequest;
+import javax.net.ssl.SSLSocketFactory;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public final class SlashEncodingContractTest {
+
+    @Rule
+    public final MockWebServer server = new MockWebServer();
+
+    private FakeoInterface proxy;
+
+    @Before
+    public void before() {
+        proxy = FeignClients.standard().createProxy(Optional.<SSLSocketFactory>absent(),
+                ImmutableSet.of("http://localhost:" + server.getPort()),
+                FakeoInterface.class);
+        server.enqueue(new MockResponse().setBody("\"foo\""));
+    }
+
+    public interface FakeoInterface {
+        @GET
+        @Path("/foo/{foo}")
+        String encoded(@PathParam("foo") String foo, @QueryParam("bar") String bar);
+    }
+
+    @Test
+    public void testEncodePathParam() throws InterruptedException {
+        proxy.encoded("slash/path", "slash/query");
+        RecordedRequest request = server.takeRequest();
+        assertThat(request.getPath(), is("/foo/slash%2Fpath?bar=slash%2Fquery"));
```

Comment:
thought the same thing except theres no easy way around this. Feign encodes and replaces query and path params then does the slash decoding to the entire resolved url: https://github.com/Netflix/feign/blob/master/core/src/main/java/feign/RequestTemplate.java#L216

My understanding is that Jersey shouldn't have a problem decoding query params though.